styx-common/src/main/java/com/spotify/styx/serialization/PersistentWorkflowInstanceState.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * -\-\-
+ * Spotify Styx Scheduler Service
+ * --
+ * Copyright (C) 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.serialization;
+
+import com.spotify.styx.state.RunState;
+import com.spotify.styx.state.RunState.State;
+import com.spotify.styx.state.StateData;
+import io.norberg.automatter.AutoMatter;
+import java.time.Instant;
+import javax.annotation.Nullable;
+
+@AutoMatter
+public interface PersistentWorkflowInstanceState {
+
+  long counter();
+
+  // TODO: remove @Nullable when all persisted active states in datastore contains the runstate
```

Comment:
TODO remove counter() as well, I guess?

Reply:
Not sure about that. Might still need it to be able to index events we write to bigtable or datastore.